sorting/external_sort.cpp

```diff
@@ -0,0 +1,187 @@
+/**
+ * C++ program to sort a file data using external sorting
+ * External sorting is required when the data being sorted do not fit into the
```

Comment:
please follow the documentation guidelines 
read the contribution.md file to see the proper header comment block each file must be accompanied with
```cpp
/**
 * @file
 * @brief < a brief description of the algorithm with a link to one source of the algorithm>
 * @details <a more detailed description of the algorithm
 *
 * @author [yourname](link to your github)
 */
 ````